src/repeat.js

```diff
@@ -94,17 +115,74 @@ export class Repeat {
     }
   }
 
-  processItemsByStrategy() {
+  getInnerCollection() {
+    let expression = unwrapExpression(this.sourceExpression);
+    if (!expression) {
+      return null;
+    }
+    return expression.evaluate(this.scope, null);
+  }
+
+  observeInnerCollection() {
+    let items = this.getInnerCollection();
+    if (items instanceof Array) {
+      this.collectionObserver = this.observerLocator.getArrayObserver(items);
+    } else if (items instanceof Map) {
+      this.collectionObserver = this.observerLocator.getMapObserver(items);
+    } else {
+      return false;
+    }
+    this.callContext = 'handleInnerCollectionChanges';
+    this.collectionObserver.subscribe(this.callContext, this);
+    return true;
+  }
+
+  observeCollection() {
     let items = this.items;
     this.collectionObserver = this.collectionStrategy.getCollectionObserver(items);
-    this.collectionStrategy.processItems(items);
     if (this.collectionObserver) {
-      this.callContext = 'handleChanges';
+      this.callContext = 'handleCollectionChanges';
       this.collectionObserver.subscribe(this.callContext, this);
     }
   }
 
-  handleChanges(collection, changes) {
+  processItemsByStrategy() {
+    if (!this.observeInnerCollection()) {
+      this.observeCollection();
+    }
+    this.collectionStrategy.processItems(this.items);
+  }
+
+  handleCollectionChanges(collection, changes) {
     this.collectionStrategy.handleChanges(collection, changes);
   }
+
+  handleInnerCollectionChanges(collection, changes) {
+    let newItems = this.sourceExpression.evaluate(this.scope, this.lookupFunctions);
+    if (newItems === this.items) {
+      return;
+    }
+    this.items = newItems;
+    this.itemsChanged();
+  }
```

Comment:
@martingust does this look right?  It works but I'm wondering if there's a more efficient way.  Some background:

When the `handleInnerCollectionChanges` method is called it means the "inner collection" (the collection that is being converted by a value converter(s)) has mutated.  Previously this "inner collection" wasn't observed.  Line 161 gets the "converted" / "actual" collection value and line 165 assigns it to the repeat's `items` property so we can essentially treat this as-if the `items` property changed.